fx.go

```diff
@@ -18,4 +18,150 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package fx // import "go.uber.org/fx"
+package fx
+
+import (
+	"io"
+	"reflect"
+
+	"github.com/pkg/errors"
+	"github.com/uber-go/tally"
+
+	"go.uber.org/dig"
+	"go.uber.org/fx/config"
+	"go.uber.org/fx/metrics"
+	"go.uber.org/fx/ulog"
+	"go.uber.org/zap"
+)
+
+// Component is a function that initializes and returns objects to be shared across the code
+type Component interface{}
+
+// Module is a building block or primary function performed by a service built on fx
+type Module interface {
+	Name() string
+	Constructor() []Component
+	Stop()
+}
+
+// Service controls the lifecycle of an fx service
+type Service struct {
+	g              *dig.Graph
+	modules        []Module
+	components     []Component
+	scopeCloser    io.Closer
```

Comment:
right now, we only have one closer of type io.Closer. I intentionally punted this to when we have more than one so we can make a decision based on that.